apps/web/src/services/dashboards/widgets/_hooks/use-widget-lifecycle.ts

```diff
@@ -174,11 +177,13 @@ export const useWidgetLifecycle = ({
     });
     watch(() => dashboardDetailState.variables, (after, before) => {
         if (!props.initiated || props.errorMode || !props.inheritOptions) return;
+        if (dashboardDetailState.widgetViewModeModalVisible && (props.widgetKey !== widgetFormState.widgetKey)) return; // disable when it's view mode
```

Comment:
description: Without this, **all** widgets will be refreshed **every time** the dashboard variable is modified in the view modal.